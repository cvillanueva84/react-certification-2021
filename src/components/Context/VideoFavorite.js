import { createContext } from 'react';

const VideoFavoriteContext = createContext({});

export default VideoFavoriteContext;
