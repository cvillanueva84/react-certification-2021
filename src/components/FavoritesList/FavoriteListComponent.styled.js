import styled from 'styled-components';

export const FavoriteListContainer = styled.div`
    display: flex;
    flex-wrap: wrap;
    width: 100%;
    justify-content: center;
`;

FavoriteListContainer.displayName = 'FavoriteListContainer'

export const NoFavoritesDiv = styled.div`
  width: 70%;
  text-align: center;
  margin-top: 10vh;
`;

export const FavoriteListEach = styled.div`
  display: flex;
  flex-wrap: wrap;
  justify-content: center;
`
FavoriteListEach.displayName = 'FavoriteListEach'

NoFavoritesDiv.displayName = 'NoFavoritesDiv'

export const VideoCardDivLight = styled.div`
  display: flex;
  flex-direction: column;
  width: 30rem;
  height: 35rem;
  border-radius: 0.4rem;
  margin: 2rem;
  border: 1px var(--light-gray) solid;
  overflow-wrap: break-word;
  transition: all 0.3s ease;
  &:hover {
      box-shadow: 10px 10px 24px 0px var(--black);
    }
    
    @media only screen and (max-width: 670px) {
        width: 80%;
    }
`;
VideoCardDivLight.displayName = 'VideoCardDivLight';

export const VideoCardDivDark= styled.div`
  display: flex;
  flex-direction: column;
  width: 30rem;
  height: 35rem;
  border-radius: 0.4rem;
  margin: 2rem;
  border: 1px var(--light-gray) solid;
  overflow-wrap: break-word;
  transition: all 0.3s ease;
  &:hover {
      -webkit-box-shadow: 5px 5px 15px 5px var(--black); 
      box-shadow: 5px 5px 15px 5px var(--light-gray) ;
    }
    
    @media only screen and (max-width: 670px) {
        width: 80%;
    }
`;
VideoCardDivDark.displayName = 'VideoCardDivDark';

export const VideoImageDiv = styled.div`
  height: 140px;
`;
VideoImageDiv.displayName = 'VideoImageDiv';

export const TitleSpan = styled.span`
  font-size: 1.6rem;
  font-weight: bold;
  margin: 2rem;
`;
TitleSpan.displayName = 'TitleSpan';

export const DescriptionSpan = styled.span`
  font-size: 1.4rem;
  margin-bottom: 2rem;
  text-align: left;
  margin: 2rem;
  `;
DescriptionSpan.displayName = 'DescriptionSpan';

