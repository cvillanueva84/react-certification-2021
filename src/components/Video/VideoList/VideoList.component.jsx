import React from 'react';
import { videoData } from '../../../youtube-videos-mock';
import { Container } from './VideoList.styled';
import VideoItem from '../VideoItem/VideoItem.component';

export default function VideoList() {
  const { items } = videoData;

  const avatar = items[0].snippet.thumbnails.default.url;

  return (
    <Container>
      {items.slice(1).map((item, id) => {
        const {
          snippet: {
            thumbnails: {
              medium: { url },
            },
          },
        } = item;

        const {
          snippet: { title },
        } = item;

        const {
          snippet: { description },
        } = item;

        return (
          <VideoItem
            key={id}
            url={url}
            title={title}
            description={description}
            avatar={avatar}
          />
        );
      })}
    </Container>
  );
}
