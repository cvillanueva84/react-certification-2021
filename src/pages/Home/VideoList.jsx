import React from 'react';
import styled from 'styled-components';
import VideoItem from './VideoItem';

const VideoListContainer = styled.div`
    width: 100%;
    height: 100vh;
    margin: 5px auto;
    display: flex;
    flex-wrap: wrap;
    justify-content: center;
    
    & a {
        text-decoration: none;
    }
`;

const VideoList = ({videos}) => {

    if (videos.length < 0 ) return null;

    return (
        <VideoListContainer>
            {videos.items.map( (item) => (
                <VideoItem item={item} key={item.etag} />
            ))}
        </VideoListContainer>
    );

};

 export default VideoList;

 