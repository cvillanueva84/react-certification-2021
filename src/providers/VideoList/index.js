export {  VideoListProvider } from './VideoList.provider';
