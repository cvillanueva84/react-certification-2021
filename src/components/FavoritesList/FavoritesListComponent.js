import React, { useContext } from 'react';
import { Link } from 'react-router-dom';
import Context from '../../providers/Theme/Theme.context';
import { VideoListContext } from '../../providers/VideoList/VideoList.provider';
import {
  DescriptionSpan, FavoriteListContainer, FavoriteListEach, NoFavoritesDiv, TitleSpan, VideoCardDivDark, VideoCardDivLight, VideoImageDiv
} from './FavoriteListComponent.styled';

const FavoritesList = () => {
  const { videosState } = useContext(VideoListContext)
  const { favoriteVideos } = videosState
  const { state } = useContext(Context);

  return (
    <FavoriteListContainer >
      {favoriteVideos.length === 0 || favoriteVideos === null ? (
        <NoFavoritesDiv>
          <h2>There are no favorite videos!</h2>
        </NoFavoritesDiv>
      ) : (
        <FavoriteListEach>
          {favoriteVideos.map((video) => (
            <Link
              key={video.etag}
              to={`/favorite-video/${video.id.videoId}`}
              style={{ textDecoration: 'none', color: '#000' }}
            >

              {state.isDark ? (
                <VideoCardDivLight>
                  <VideoImageDiv
                    style={{
                      backgroundImage: `url(${video.snippet.thumbnails.medium.url})`,
                      backgroundSize: 'cover',
                      backgroundRepeat: 'no-repeat',
                      backgroundPosition: 'center',
                      borderTopRightRadius: '0.4rem',
                      borderTopLeftRadius: '0.4rem',
                    }}
                  ></VideoImageDiv>
                  <TitleSpan>{video.snippet.title}</TitleSpan>
                  <DescriptionSpan>
                    {video.snippet.description
                      ? video.snippet.description.slice(0, 100)
                      : 'watch video...'}
                  </DescriptionSpan>
                </VideoCardDivLight>
              ) : (
                <VideoCardDivDark>
                  <VideoImageDiv
                    style={{
                      backgroundImage: `url(${video.snippet.thumbnails.medium.url})`,
                      backgroundSize: 'cover',
                      backgroundRepeat: 'no-repeat',
                      backgroundPosition: 'center',
                      borderTopRightRadius: '0.4rem',
                      borderTopLeftRadius: '0.4rem',
                    }}
                  ></VideoImageDiv>
                  <TitleSpan>{video.snippet.title}</TitleSpan>
                  <DescriptionSpan>
                    {video.snippet.description
                      ? video.snippet.description.slice(0, 100)
                      : 'watch video...'}
                  </DescriptionSpan>
                </VideoCardDivDark>
              )}
            </Link>
          ))}
        </FavoriteListEach>
      )}
    </FavoriteListContainer>
  );
};

export default FavoritesList;
