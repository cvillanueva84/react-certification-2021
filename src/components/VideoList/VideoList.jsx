import React, { useEffect, useState } from 'react';
import axios from 'axios';
import { DivTitle, Container, Fakegrid } from './VideoList.styled';
import Card from '../Card/Card';

const baseURL =
  'https://gist.githubusercontent.com/jparciga/1d4dd34fb06ba74237f8966e2e777ff5/raw/f3af25f1505deb67e2cc9ee625a633f24d8983ff/youtube-videos-mock.json';

const VideoList = () => {
  const [videos, setVideos] = useState([]);

  useEffect(() => {
    async function getVideos() {
      const {
        data: { items },
      } = await axios.get(baseURL);
      const aux = [...items];
      aux.shift();
      setVideos(aux);
      // console.log(data.items.map((el) => el.snippet.description));
    }
    getVideos();
  }, []);

  return (
    <>
      <DivTitle>
        <h2>Welcome to the Challenge!</h2>
      </DivTitle>
      <Container>
        <Fakegrid>
          {videos.map((video) => (
            <Card
              title={video.snippet.title}
              description={video.snippet.description}
              link={video.snippet.thumbnails.high.url}
              key={video.snippet.publishedAt}
            />
          ))}
        </Fakegrid>
      </Container>
    </>
  );
};

export default VideoList;
