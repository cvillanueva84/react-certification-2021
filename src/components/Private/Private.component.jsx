/* eslint-disable react/jsx-props-no-spreading */ 
import React from 'react';
import PropTypes from 'prop-types';
import { Route, Redirect } from 'react-router-dom';

import { useAuth } from '../../providers/Auth';

function Private({ children, ...rest }) {
  const { authenticated } = useAuth();

  return (
    <Route {...rest} render={() => (authenticated ? children : <Redirect to="/" />)} />
  );
}
Private.propTypes = {
  children: PropTypes.node.isRequired
}

export default Private;
