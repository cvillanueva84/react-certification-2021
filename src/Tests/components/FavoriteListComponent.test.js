import { mount } from 'enzyme';
import React from 'react';
import { MemoryRouter } from 'react-router';
import FavoritesList from '../../components/FavoritesList/FavoritesListComponent';
import Context from '../../providers/Theme/Theme.context';
import { VideoListContext } from '../../providers/VideoList/VideoList.provider';
import { videosState } from '../mockData/videosList';
import AuthProvider from '../../providers/Auth';
import Private from '../../components/Private/PrivateComponent';


describe('Tests of the FavoriteListComponent', () => {
    const state = {
        isDark: true
    }
    const authenticated = true
    const wrapper = mount(
        <AuthProvider
            value={{authenticated}}
        >
            <Context.Provider
                value={{state}}
                >
                <VideoListContext.Provider
                    value={{
                        videosState
                    }}
                    >
                <MemoryRouter>
                    < FavoritesList/>
                </MemoryRouter>
                </VideoListContext.Provider>
            </Context.Provider>
        </AuthProvider>
    )
    test('should render the component correctly', () => {
        expect(wrapper).toMatchSnapshot()
    })
    

    test('should render a message if no videos are added to favorites', () => {
        if (videosState.length === 0) {
            const warningMsg = wrapper.find('h2')
            expect(warningMsg.length).toBe(1)
        }
    })
    
    test('should not show this component if the user is not authenticated', () => {
        const state = {
            isDark: true
        }
        const authenticated = false
        const wrapper = mount(
            <AuthProvider
                value={{authenticated}}
            >
                <Context.Provider
                    value={{state}}
                    >
                    <VideoListContext.Provider
                        value={{
                            videosState
                        }}
                        >
                    <MemoryRouter>
                        <Private>
                            < FavoritesList/>
                        </Private>
                    </MemoryRouter>
                    </VideoListContext.Provider>
                </Context.Provider>
            </AuthProvider>
        )
        const favoriteListComponent = wrapper.find('FavoriteListContainer')
        expect(favoriteListComponent.exists()).toBe(false)
    })
    
    test('should display the title of the video', () => {
        const title = wrapper.find('TitleSpan').at(0)
        expect(title.text()).toEqual(videosState.posts[1].snippet.title)
    })
    
    test('should show a custom message if the video does not have a description', () => {
        const description = wrapper.find('DescriptionSpan').at(0)
        if (description.text().length === '') {
            expect(description.text()).toBe('watch video...')
        } else {
            expect(description.text()).toBe(videosState.posts[1].snippet.description.slice(0, 100))
        }
    })
    
})


