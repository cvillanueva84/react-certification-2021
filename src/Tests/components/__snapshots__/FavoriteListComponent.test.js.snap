// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Tests of the FavoriteListComponent should render the component correctly 1`] = `
<div
  className="sc-bdnxRM eWDhhJ"
>
  <div
    className="sc-dlnjwi khqPoZ"
  >
    <a
      href="/favorite-video/undefined"
      onClick={[Function]}
      style={
        Object {
          "color": "#000",
          "textDecoration": "none",
        }
      }
    >
      <div
        className="sc-hKFxyN lmYhKb"
      >
        <div
          className="sc-jSFjdj bqukUU"
          style={
            Object {
              "backgroundImage": "url(https://i.ytimg.com/vi/HYyRZiwBWc8/mqdefault.jpg)",
              "backgroundPosition": "center",
              "backgroundRepeat": "no-repeat",
              "backgroundSize": "cover",
              "borderTopLeftRadius": "0.4rem",
              "borderTopRightRadius": "0.4rem",
            }
          }
        />
        <span
          className="sc-gKAaRy gWKPHz"
        >
          Wizeline Guadalajara | Bringing Silicon Valley to Mexico
        </span>
        <span
          className="sc-iCoGMd gIgibD"
        >
          Wizeline continues to offer a Silicon Valley culture in burgeoning innovation hubs like Mexico and V
        </span>
      </div>
    </a>
    <a
      href="/favorite-video/undefined"
      onClick={[Function]}
      style={
        Object {
          "color": "#000",
          "textDecoration": "none",
        }
      }
    >
      <div
        className="sc-hKFxyN lmYhKb"
      >
        <div
          className="sc-jSFjdj bqukUU"
          style={
            Object {
              "backgroundImage": "url(https://i.ytimg.com/vi/nmXMgqjQzls/mqdefault.jpg)",
              "backgroundPosition": "center",
              "backgroundRepeat": "no-repeat",
              "backgroundSize": "cover",
              "borderTopLeftRadius": "0.4rem",
              "borderTopRightRadius": "0.4rem",
            }
          }
        />
        <span
          className="sc-gKAaRy gWKPHz"
        >
          Video Tour | Welcome to Wizeline Guadalajara
        </span>
        <span
          className="sc-iCoGMd gIgibD"
        >
          Follow Hector Padilla, Wizeline Director of Engineering, for a lively tour of our office.

In 2018, 
        </span>
      </div>
    </a>
  </div>
</div>
`;
